File: server/ruby/app/infrastructure/mappers/payment_details/address_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class AddressMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::Address.new
          dto.additional_info = response.additional_info
          dto.city = response.city
          dto.country_code = response.country_code
          dto.house_number = response.house_number
          dto.state = response.state
          dto.street = response.street
          dto.zip = response.zip

          dto
        end
      end
    end
  end
end

File: server/nodejs/src/infrastructure/mappers/payment-details/address-mapper.ts
import type { Domain } from 'onlinepayments-sdk-nodejs';
import { Address } from '../../../business/domain/payments/payment-details/address';

type AddressSdk = Domain.Address;

export const AddressMapper = {
    fromSdkResponse: (response?: AddressSdk | null): Address | undefined => {
        if (!response) {
            return undefined;
        }

        return {
            city: response?.city,
            countryCode: response?.countryCode,
            additionalInfo: response?.additionalInfo,
            houseNumber: response?.houseNumber,
            state: response?.state,
            street: response?.street,
            zip: response?.zip,
        };
    },
};
